### PR TITLE
DLSV2-513 Create a nominated supervisor role

### PR DESCRIPTION
**Pull request by daniel-manta-hee**
### JIRA link
https://hee-dls.atlassian.net/browse/DLSV2-513

### Description
Added migration, updated tests, views, controller methods and helpers.

#### Administrator/EditAdminRoles.cshtml
image

#### Delegates/PromoteToAdmin/Index.cshtml
image

-----
### Developer checks

I have:
- Run Data and Web tests.
- Checked Add staff members not available in `Supervisor/MyStaffList.cshtml` when user is nominated supervisor.
- Checked that Enrol on self assessments not available in `Supervisor/DelegateProfileAssessments.cshtml` when user is nominated supervisor.
- Checked that new role Is Nominated Supervisor is available in `Administrator/EditAdminRoles.cshtml`.
- Checked that new role Is Nominated Supervisor is available in `Delegates/PromoteToAdmin/Index.cshtml`.

### CODE REVIEW

File: DigitalLearningSolutions.Web/ViewModels/Supervisor/MyStaffListViewModel.cs [hunk 1]
```diff
@@ -20,6 +22,7 @@ public class MyStaffListViewModel : BaseSearchablePageViewModel
             int page
         ) : base(searchString, page, false, sortBy, sortDirection, searchLabel: "Search administrators")
         {
+            IsNominatedSupervisor = adminUser?.IsNominatedSupervisor ?? false;
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Web/ViewModels/Supervisor/MyStaffListViewModel.cs:25]: Given that the adminUser may, in theory, be a Supervisor as well as a NominatedSupervisor, we should handle that here (`IsNominatedSupervisor` should be `false` if `adminUser.IsSupervisor` is `true`).

File: DigitalLearningSolutions.Web/Views/Supervisor/MyStaffList.cshtml [hunk 2]
```diff
@@ -72,34 +72,37 @@
...
+          @Html.HiddenFor(m => m.SearchString)
+          @Html.HiddenFor(m => m.SortBy)
+          @Html.HiddenFor(m => m.SortDirection)
+          @Html.HiddenFor(m => m.Page)
+        </form>
+      </div>
     </div>
-  </div>
+  }
   <div class="nhsuk-grid-row">
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Web/Views/Supervisor/MyStaffList.cshtml:106]: `Add multiple staff members` interface elements should also be hidden for `IsNominatedSupervisor`

**Review by kevwhitt-hee — CHANGES_REQUESTED** [review]
A few tweaks requested in comments and one question.

File: DigitalLearningSolutions.Web/Controllers/SupervisorController/SupervisorController.cs [hunk 3]
```diff
@@ -1,61 +1,65 @@
...
+           ILogger<SupervisorController> logger,
+           IConfiguration config)
+        {
+            this.supervisorService = supervisorService;
+            this.commonService = commonService;
+            this.frameworkNotificationService = frameworkNotificationService;
+            this.selfAssessmentService = selfAssessmentService;
+            this.configService = configService;
+            this.centreCustomPromptsService = centreCustomPromptsService;
+            this.userDataService = userDataService;
```
[the remark this answers is not in the document]
Comment (daniel-manta-hee) [reply DigitalLearningSolutions.Web/Controllers/SupervisorController/SupervisorController.cs:40]: I have added userDataService that is used in Supervisor.cs line 46 and 172. Apart from that, I see `selfAssessmentService`, `centreCustomPromptsService` are being used (e.g. if you right click `private readonly ISelfAssessmentService selfAssessmentService;` and click on "Find All References") and `config` service not being used.

File: DigitalLearningSolutions.Web/ViewModels/Supervisor/MyStaffListViewModel.cs (hunk 1 above)
Comment (daniel-manta-hee) [reply DigitalLearningSolutions.Web/ViewModels/Supervisor/MyStaffListViewModel.cs]: Added [this commit](https://github.com/TechnologyEnhancedLearning/DLSV2/pull/982/commits/0d0be3c7edd21986c22f9feb24c0abe6d74d9ba2).

**Review by kevwhitt-hee — APPROVED** [review]

**Pull request merged**